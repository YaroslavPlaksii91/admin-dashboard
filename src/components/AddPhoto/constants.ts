export const IMAGE_SIZE = Object.freeze({
  SMALL: 44,
  BIG: 60,
});
