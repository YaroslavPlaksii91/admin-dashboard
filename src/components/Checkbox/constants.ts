export const CHECKBOX_TEST_ID = 'checkbox';
