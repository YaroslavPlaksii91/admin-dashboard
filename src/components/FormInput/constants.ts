export const TOGGLE_BUTTON_TEST_ID = 'toggleButton';
