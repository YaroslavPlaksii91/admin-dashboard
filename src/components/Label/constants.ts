export const LABEL_TYPES = Object.freeze({
  ROUND: 'round',
  SQUARE: 'square',
});

export const LABEL_COLORS = Object.freeze({
  DEFAULT: 'default',
  SUCCESS: 'success',
  WARNING: 'warning',
  ERROR: 'error',
});
