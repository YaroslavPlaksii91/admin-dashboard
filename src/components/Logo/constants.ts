export const LOGO_TYPES = Object.freeze({
  VERTICAL: 'vertical',
  HORIZONTAL: 'horizontal',
});
