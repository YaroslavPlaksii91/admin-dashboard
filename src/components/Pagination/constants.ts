export const ROWS_PER_PAGE = 8;
export const ROWS_PER_PAGE_OPTIONS = [8, 12, 16, 20];
