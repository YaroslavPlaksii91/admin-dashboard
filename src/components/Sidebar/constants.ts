export const USERS_PAGES = Object.freeze([
  Object.freeze({
    name: 'Overview',
    icon: '/src/assets/icons/sprite.svg#icon-overview',
    link: '/overview',
  }),
  Object.freeze({
    name: 'Tickets',
    icon: '/src/assets/icons/sprite.svg#icon-tickets',
    link: '/tickets',
  }),
  Object.freeze({
    name: 'Ideas',
    icon: '/src/assets/icons/sprite.svg#icon-ideas',
    link: '/ideas',
  }),
  Object.freeze({
    name: 'Contacts',
    icon: '/src/assets/icons/sprite.svg#icon-contacts',
    link: '/contacts',
  }),
  Object.freeze({
    name: 'Agents',
    icon: '/src/assets/icons/sprite.svg#icon-agents',
    link: '/agents',
  }),
  Object.freeze({
    name: 'Articles',
    icon: '/src/assets/icons/sprite.svg#icon-articles',
    link: '/articles',
  }),
]);

export const USERS_SETTINGS = Object.freeze([
  Object.freeze({
    name: 'Settings',
    icon: '/src/assets/icons/sprite.svg#icon-settings',
    link: '/settings',
  }),
  Object.freeze({
    name: 'Subscription',
    icon: '/src/assets/icons/sprite.svg#icon-subscription',
    link: '/subscription',
  }),
]);
