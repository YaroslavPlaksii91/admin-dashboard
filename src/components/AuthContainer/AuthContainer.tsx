import { FC } from 'react';

import { AuthContainerProps } from './types';
import { AUTH_CONTAINER_TEST_ID } from './constants';
import { Logo } from '@components/Logo/Logo';

import styles from './AuthContainer.module.css';

export const AuthContainer: FC<AuthContainerProps> = ({ children }) => {
  return (
    <div className={styles.wrapper} data-testid={AUTH_CONTAINER_TEST_ID}>
      <Logo />
      {children}
    </div>
  );
};
