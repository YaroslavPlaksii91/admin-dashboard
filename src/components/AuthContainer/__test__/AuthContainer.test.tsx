import { MemoryRouter } from 'react-router-dom';
import { cleanup, render, screen } from '@testing-library/react';

import { AUTH_CONTAINER_TEST_ID } from '../constants';
import { AuthContainer } from '../AuthContainer';

describe('AuthContainer component', () => {
  afterEach(cleanup);

  const MockedChildren = () => <div>Mocked Children</div>;

  it('should render AuthContainer component', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <AuthContainer>
          <MockedChildren />
        </AuthContainer>
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should be in the document', () => {
    render(
      <MemoryRouter>
        <AuthContainer>
          <MockedChildren />
        </AuthContainer>
      </MemoryRouter>,
    );

    expect(screen.getByTestId(AUTH_CONTAINER_TEST_ID)).toBeInTheDocument();
  });

  it('children should be in the document', () => {
    render(
      <MemoryRouter>
        <AuthContainer>
          <MockedChildren />
        </AuthContainer>
      </MemoryRouter>,
    );

    expect(screen.getByText('Mocked Children')).toBeInTheDocument();
  });
});
