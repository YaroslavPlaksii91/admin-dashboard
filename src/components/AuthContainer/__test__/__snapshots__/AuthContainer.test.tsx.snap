// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthContainer component should render AuthContainer component 1`] = `
<DocumentFragment>
  <section
    data-testid="authContainer"
  >
<<<<<<< HEAD
    <div
      class=""
    >
      <a
        href="/"
=======
    <a
      href="/"
    >
      <div
        class=""
>>>>>>> develop
      >
        <svg
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          focusable="false"
          role="img"
          viewBox="0 0 24 24"
        >
          <use
            href="/src/assets/icons/sprite.svg#icon-logo"
          />
          <title>
            Logo
          </title>
        </svg>
        <p
<<<<<<< HEAD
          class="MuiTypography-root MuiTypography-h3 css-14qosaq-MuiTypography-root"
        >
          Dashboard Kit
        </p>
      </a>
    </div>
=======
          class="MuiTypography-root MuiTypography-h3 css-army73-MuiTypography-root"
        >
          Dashboard Kit
        </p>
      </div>
    </a>
>>>>>>> develop
    <div>
      Mocked Children
    </div>
  </section>
</DocumentFragment>
`;
